fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -317,6 +329,21 @@ public <R> CompletableFuture<R> apply(Function<FDBRecordContext, CompletableFutu
             });
         }
 
+        @Override
+        @SuppressWarnings("PMD.CloseResource") // closed in a future
+        public <R> CompletableFuture<R> applyInRecoveryPath(Function<FDBRecordContext, CompletableFuture<R>> function) {
```

Comment:
We should add tests that this does what we want, namely, if we have a bunch of threads using the same `AgilityContext`, with a low time limit, it should commit frequently, and once we call `abortAndClose` nothing else should be committed, and everything should be aborted.
And we then call `applyInRecoverContext` and that works. I think if you have an `AtomicBoolean` that is written from the other threads, and gets changed to `true` right after closing, and then assert that `true` is never written.
It's not perfect, but better than nothing (although perhaps not better than something you can think of)